doc/kci_testsuite.md

```diff
@@ -0,0 +1,300 @@
+
+5. The `test_overlay` option is used to create a specific directory layout inside the rootfs image. In our case:
+
+```
+$ tree jenkins/debian/debos/overlays/v4l2/
+jenkins/debian/debos/overlays/v4l2/
+└── usr
+    └── bin
+        └── v4l2-parser.sh
+
+```
+   `/usr/bin/v4l2-parser.sh` will be available in the `buster-v4l2` rootfs image. These overlay scripts are normally run from LAVA job to execute some tests and parse the results.
+    This sample [LAVA job definition](https://lava.collabora.co.uk/scheduler/job/2446126/definition#defline120) shows how `v4l2-parser.sh` can be used.
+
+
+6. Hopefully now you should have some basic idea about how to add a test suite to a rootfs and how it can be used in LAVA environment.
+
+### How to add new testsuite to a buster rootfs image
+
+For this example, we will be using `btrfs-progs` test-sute.
+
+1. Let's first modify `rootfs-configs.yaml` for `buster-btrfs` with appropriate entires. 
+
+
+```
+  buster-btrfs:
+    rootfs_type: debos
+    debian_release: buster
+    arch_list:
+      - amd64
+    extra_packages:
+      - git 
+      - autoconf 
+      - automake 
+      - gcc 
+      - make 
+      - pkg-config 
+      - e2fslibs-dev 
+      - libblkid-dev 
+      - zlib1g-dev 
+      - liblzo2-dev 
+      - asciidoc 
+      - xmlto 
+      - libzstd-dev
+      - python3-setuptools
+      - python3.7-dev
+    script: "scripts/buster-btrfs.sh"
+    test_overlay: "overlays/btrfs"
+```
+
+
+2. We will look into last three config entries namely, `extra_packages`, `script` and `test_overlay`.
+   `extra_packages` as name suggests it will add the listed packages from final rootfs image.
+
+3. Remember `script` path is relative to `kci_rootfs build --data-path` value. By default its `jenkins/debian/debos/script`
+   Here we need add new script named `buster-btrfs.sh`. Its a simple script to download the repo and build and install 
+   the btrfs-progs from source. 
+
+
+```
+$ cat jenkins/debian/debos/script/buster-btrfs.sh
+
+git clone https://github.com/kdave/btrfs-progs.git /btrfs-progs
+cd /btrfs-progs
+./autogen.sh && ./configure
+make
+make install
+```
+
+
+4. `test_overlay` option is used to create specific directory layout inside the rootfs image. In our case, 
+
+```
+$ tree jenkins/debian/debos/overlays/btrfs/
+jenkins/debian/debos/overlays/btrfs/
+└── usr
+    └── bin
+        └── btrfs-verify-cli.sh
+```
+
+   `/usr/bin/btrfs-verify-cli.sh` will be available on `buster-btrfs` rootfs image. These overlay scripts normally ran from LAVA job.
+
+```
+$ cat jenkins/debian/debos/overlays/btrfs/btrfs-verify-cli.sh
+
+cd  /btrfs-progs && TEST=001\* ./tests/cli-tests.sh
+```
+
+5. Make sure execute permission is applied on the scripts. 
+
+```
+chmod +x jenkins/debian/debos/overlays/btrfs/usr/bin/btrfs-verify-cli.sh
+chmod +x jenkins/debian/debos/script/buster-btrfs.sh
+```
+
+6. Now build `buster-btrfs` rootfs image using kci_rootfs. 
```

Comment:
It's probably worth adding a sample command here, something like this (untested):
```sh
./kci_rootfs build --arch=amd64 --rootfs-config=buster-btrfs
```